### DISCUSSION TITLE
Does serde_with support post processor, or there is another way to do this?

### DISCUSSION THREAD

**Discussion by Magicloud** [jonasbb/serde_with#796] [Q&A] [answered]
I need to use a web interface giving a JSON object. Converting to Rust, would be:

```
struct Obj {
  field_a: Vec<String>,
  field_b: Vec<String>,
  field_c: Option<Vec<String>>,
}
```

The interface actually uses empty Vec when there is nothing for the field, but for reason unknown, the author made `field_c` `null` when there is nothing. It looks really inconsistent.

So I wonder if I could do something like:

```
#[serde_as(as = "Option<Vec<String>>", de-post-processor = "unwrap_or_default")]
field_c: Vec<String>
```

**Comment by jonasbb** [comment, marked as answer]
I think your use case is covered by [`DefaultOnNull`](https://docs.rs/serde_with/latest/serde_with/struct.DefaultOnNull.html).

A general post-processing like this is not really feasible outside `serde_derive`. Functions are not generic types, so you cannot have a `PostProcess<unwrap_or_default>` or similar.

**Reply by Magicloud** [reply]
Thank you.
